### PR TITLE
CORDA-578 CordaX500Name JAX-RS parameter converter

### PR DESCRIPTION
**Pull request by m4ksio**
CordaX500Name can be used as a @QueryParam in JAX-RS services

### CODE REVIEW

File: webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class CordaConverterProvider : ParamConverterProvider {
+
+    override fun <T : Any> getConverter(rawType: Class<T>?, genericType: Type?, annotations: Array<out Annotation>?): ParamConverter<T>? {
+        if (rawType == CordaX500Name::class.java) {
+            @Suppress("UNCHECKED_CAST")
+            return CordaX500NameConverter as ParamConverter<T>
+        }
+        return null
+    }
+}
```
Comment (andr3ej) [inline webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt]: please minimise the number of blank lines in this file (and run the formatter)

File: webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    override fun fromString(value: String): CordaX500Name {
+        return CordaX500Name.parse(value)
+    }
+
+}
+
+@Provider
+class CordaConverterProvider : ParamConverterProvider {
+
+    override fun <T : Any> getConverter(rawType: Class<T>?, genericType: Type?, annotations: Array<out Annotation>?): ParamConverter<T>? {
```
Comment (andr3ej) [inline webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt]: most of these question marks are the java->kotlin importer being pessimistic. can you retain them only on the params that are actually nullable

File: webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    override fun fromString(value: String): CordaX500Name {
+        return CordaX500Name.parse(value)
+    }
+
+}
+
+@Provider
+class CordaConverterProvider : ParamConverterProvider {
```
Comment (andr3ej) [inline webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt]: can this be an object?

File: webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+object CordaX500NameConverter : ParamConverter<CordaX500Name> {
+
+    override fun toString(value: CordaX500Name): String {
+        return value.toString()
+    }
+
+    override fun fromString(value: String): CordaX500Name {
+        return CordaX500Name.parse(value)
+    }
```
Comment (andr3ej) [inline webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt]: these 2 methods are simple enough that the short form
fun name(...) = expression
is appropriate

File: webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt (hunk 1 above)
Comment (m4ksio) [reply webserver/src/main/kotlin/net/corda/webserver/converters/Converters.kt]: Reformatted as requested

**Comment by josecoll** [comment]
Please ask @mikehearn to approve this as my understanding is that we do not make changes to accommodate Web Applications otherwise we end up having to support them.

**Comment by mikehearn** [comment]
Hmm, no, we have no such policy. I might not like web apps myself for a variety of reasons but we surely do support people creating them!

**Pull request merged**